tests/environment-hook.bats

```diff
@@ -13,14 +13,17 @@ load '/usr/local/lib/bats/load.bash'
   export BUILDKITE_PLUGIN_VAULT_SECRETS_SERVER=https://vault_svr_url
   export BUILDKITE_PLUGIN_VAULT_SECRETS_DUMP_ENV=true
   export BUILDKITE_PIPELINE_SLUG=testpipe
-  export TESTDATA=`echo MY_SECRET=fooblah | base64`
+  export TESTDATA='MY_SECRET=fooblah'
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_SECRET_ENV="abcde12345"
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_ROLE_ID=buildkite
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD=approle
 
   stub vault \
-    'auth -address=https://vault_svr_url - : echo Successfully authenticated. You are now logged in' \
-    "list -address=https://vault_svr_url -format=yaml data/buildkite/testpipe : exit 0 " \
-    "list -address=https://vault_svr_url -format=yaml data/buildkite : echo -e '- env'" \
-    "list -address=https://vault_svr_url -format=yaml data/buildkite/env : echo -e '- MY_SECRET'" \
-    "read -address=https://vault_svr_url -field=value data/buildkite/env/MY_SECRET : echo ${TESTDATA}"
+    "write -field=token -address=https://vault_svr_url auth/approle/login role_id=buildkite secret_id=\"${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_SECRET_ENV}\" : echo 'Successfully authenticated with Role ID ${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_ROLE_ID}'" \
```

Comment:
You can make this stub more generic (and shorter) with the following:

```suggestion
    "write -field=token \* auth/approle/login \* \* : echo 'Successfully authenticated with \${5} \${6}" \
```

Which is basically accepting any parameters in the 3rd, 5th and 6th positions and using the last two in the output so you can later make assertions on the output

Reply:
I've actually completely removed the auth section of these tests because it's not needed to test the functionality of evaluating `env` received from Vault. This is good to know though!